File: be/app/dto/cargo_dto.py
from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
from typing import Optional


class CargoItem(BaseModel):
    item: str
    ea: int


class CargoRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    projectName: str
    from_: str = Field(alias="from")
    stopover: Optional[str] = None
    to: str
    items: list[CargoItem]
    atd: Optional[datetime] = None
    eta: Optional[datetime] = None
    ata: Optional[datetime] = None


class CargoResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    projectName: str
    from_: str = Field(alias="from")
    stopover: Optional[str] = None
    to: str
    items: list[CargoItem]
    atd: Optional[datetime] = None
    eta: Optional[datetime] = None
    ata: Optional[datetime] = None
    add_at: datetime
    updated_at: datetime

File: fe/src/dto/cargoDto.js
export const EMPTY_CARGO_ITEM = {
  item: '',
  ea: 1,
}

export const EMPTY_CARGO_FORM = {
  projectName: '',
  from: '',
  stopover: '',
  to: '',
  items: [{ ...EMPTY_CARGO_ITEM }],
  atd: '',
  eta: '',
  ata: '',
}

export const toCargoPayload = (form) => ({
  projectName: form.projectName,
  from: form.from,
  stopover: form.stopover || null,
  to: form.to,
  items: form.items,
  atd: form.atd || null,
  eta: form.eta || null,
  ata: form.ata || null,
})

export const toCargoForm = (cargo) => ({
  projectName: cargo.projectName,
  from: cargo.from,
  stopover: cargo.stopover ?? '',
  to: cargo.to,
  items: cargo.items.map((item) => ({ ...item })),
  atd: cargo.atd ?? '',
  eta: cargo.eta ?? '',
  ata: cargo.ata ?? '',
})
